src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py

```diff
@@ -46,7 +47,7 @@
 """
 
 
-class StableDiffusionPipeline(DiffusionPipeline):
+class StableDiffusionPipeline(DiffusionPipeline, TextualInversionMixin):
```

Comment:
Nice! Can we add the mixin to the `DiffusionPipeline` class? 

If I'm reading the textual inversion mixin correctly it's pretty general and should work on any class where `self.tokenizer` and `self.text_encoder` have the expected functionality. 

I would recommend doing a check that the two variables exist and are the correct types in the mixin's exposed methods.

i.e.

```py
if not hasattr(self, "tokenizer") or not isinstance(self.tokenizer, PreTrainedTokenizer):
    raise ValueError(f"{self.__class__.__name__} requires `self.tokenizer` of type `PreTrainedTokenizer` for calling <x>")

if not hasattr(self, "text_encoder") or not isinstance(self.text_encoder, PreTrainedModel):
    raise ValueError(f"{self.__class__.__name__} requires `self.text_encoder` of type `PreTrainedModel` for calling <x>")
```

Reply:
I've added this.